tests/unittest/_torch/visual_gen/test_qwen_image_registry.py

```
@@ -0,0 +1,96 @@
+    assert hasattr(model, "img_in")
+    assert hasattr(model, "txt_in")
+    assert hasattr(model, "txt_norm")
+    assert hasattr(model, "norm_out")
+    assert hasattr(model, "proj_out")
+    assert hasattr(model, "pos_embed")
+
+
+def test_transformer_load_weights_detects_mismatch():
+    """load_weights should surface a clear RuntimeError on a bad dict."""
+    model = QwenImageTransformer2DModel(model_config=None, num_layers=2)
+    with pytest.raises(RuntimeError, match="Missing keys|Unexpected keys"):
```

Ruff (0.15.11)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)